lib/Doctrine/DBAL/Driver/PDOConnection.php

```diff
@@ -120,7 +125,19 @@ public function quote($input, $type = \PDO::PARAM_STR)
      */
     public function lastInsertId($name = null)
     {
-        return parent::lastInsertId($name);
+        if (null !== $name) {
+            return parent::lastInsertId($name);
+        }
+
+        $lastInsertId = parent::lastInsertId($name);
+
+        // The last insert ID is reset to "0" in certain situations by some implementations,
+        // therefore we keep the previously set insert ID locally.
+        if ($lastInsertId) {
```

Comment:
These are just edge cases that should really not be considered, IMO, but it's good to comment on them, at least.